src/coreComponents/constitutive/solid/DruckerPrager.hpp

```diff
@@ -0,0 +1,449 @@
+   */
+  DruckerPragerUpdates( arrayView1d< real64 const > const & friction,
+                        arrayView1d< real64 const > const & dilation,
+                        arrayView1d< real64 const > const & hardening,
+                        arrayView2d< real64 > const & newCohesion,
+                        arrayView2d< real64 > const & oldCohesion,
+                        arrayView1d< real64 const > const & bulkModulus,
+                        arrayView1d< real64 const > const & shearModulus,
+                        arrayView3d< real64, solid::STRESS_USD > const & newStress,
+                        arrayView3d< real64, solid::STRESS_USD > const & oldStress ):
+    ElasticIsotropicUpdates( bulkModulus, shearModulus, newStress, oldStress ),
+    m_friction( friction ),
+    m_dilation( dilation ),
+    m_hardening( hardening ),
+    m_newCohesion( newCohesion ),
+    m_oldCohesion( oldCohesion )
+  {}
+
+  /// Default copy constructor
+  DruckerPragerUpdates( DruckerPragerUpdates const & ) = default;
+
+  /// Default move constructor
+  DruckerPragerUpdates( DruckerPragerUpdates && ) = default;
+
+  /// Deleted default constructor
+  DruckerPragerUpdates() = delete;
+
+  /// Deleted copy assignment operator
+  DruckerPragerUpdates & operator=( DruckerPragerUpdates const & ) = delete;
+
+  /// Deleted move assignment operator
+  DruckerPragerUpdates & operator=( DruckerPragerUpdates && ) =  delete;
+
+  /// Use the uncompressed version of the stiffness bilinear form
+  using DiscretizationOps = SolidModelDiscretizationOpsFullyAnisotroipic; // TODO: typo in anistropic (fix in DiscOps PR)
+
+  // Bring in base implementations to prevent hiding warnings
+  using ElasticIsotropicUpdates::smallStrainUpdate;
+
+  GEOSX_HOST_DEVICE
+  virtual void smallStrainUpdate( localIndex const k,
+                                  localIndex const q,
+                                  real64 const ( &strainIncrement )[6],
+                                  real64 ( &stress )[6],
+                                  real64 ( &stiffness )[6][6] ) const override final;
+
+  GEOSX_HOST_DEVICE
+  virtual void smallStrainUpdate( localIndex const k,
+                                  localIndex const q,
+                                  real64 const ( &strainIncrement )[6],
+                                  real64 ( &stress )[6],
+                                  DiscretizationOps & stiffness ) const final;
+
+private:
+  /// A reference to the ArrayView holding the friction angle for each element.
+  arrayView1d< real64 const > const m_friction;
+
+  /// A reference to the ArrayView holding the dilation angle for each element.
+  arrayView1d< real64 const > const m_dilation;
+
+  /// A reference to the ArrayView holding the hardening rate for each element.
+  arrayView1d< real64 const > const m_hardening;
+
+  /// A reference to the ArrayView holding the new cohesion for each integration point
+  arrayView2d< real64 > const m_newCohesion;
+
```

Comment:
Why cohesion is set for each quadrature point when other parameters such as friction, dilation, hardening are for each element? What is the different in their nature?